### DISCUSSION TITLE
How do I use BinaryDelta tool when I have added the package dependency not through Xcode's GUI

### DISCUSSION THREAD

**Discussion by nishaanth-ms-21155** [sparkle-project/Sparkle#2594] [Q&A] [answered]
I have added the sparkle package dependency through xcode command to generate the binary in a folder. How do I use sparkle tools like BinaryDelta and generate_keys

**Comment by zorgiepoo** [comment, marked as answer]
If you're using Swift Package Manager, it should download the tools in the artifacts directory. If you're using Xcode (most people that use SPM are) there is a way to find it documented at https://sparkle-project.org/documentation/

Otherwise you can download the [latest Sparkle distribution](https://github.com/sparkle-project/Sparkle/releases) yourself and use the tools side by side with your package manager.

I suggest trying to use `generate_appcast` first before trying to use `BinaryDelta` directly.
